ledger/src/blockstore.rs

```diff
@@ -1789,7 +1805,11 @@ impl Blockstore {
         }
 
         let slot_meta_cf = self.db.column::<cf::SlotMeta>();
-        let slot_meta = slot_meta_cf.get(slot)?;
+        let mut slot_meta = slot_meta_cf.get(slot)?;
+        if slot_meta.is_none() {
+            self.reconcile_shreds(slot, start_index);
```

Comment:
@carllin wdyt?

Reply:
hmmm doesn't seem to cover all cases. You can have multiple batches of data inserts for the same slot, so the SlotMeta can exist and still be out of sync with the data shreds. 

For instance, the first batch of inserts succeeds, but then the second batch crashes in between the data insert and SlotMeta update.